com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineAssetEditor.cs

```diff
@@ -295,6 +293,80 @@ void DrawQualitySettings()
             EditorGUILayout.EndFoldoutHeaderGroup();
         }
 
+        protected override void OnHeaderGUI()
+        {
+            // Default Header
+            base.OnHeaderGUI();
+
+            // New button in header to assign asset
+            EditorGUILayout.BeginVertical();
+
+            if (GUILayout.Button("Assign to Project Settings - Graphics"))
+            {
+                var asset = (RenderPipelineAsset)target;
+                var pipelineAsset = GraphicsSettings.defaultRenderPipeline;
+                string message = "";
+                string buttonText = "";
+                if (pipelineAsset == null)
+                {
+                    message = $"This will assign {asset.name} to the current Graphics Settings.\nRight now nothing is assigned. Do you want to assign this asset?";
+                    buttonText = "Assign";
+                }
+                else
+                {
+                    message = $"This will assign {asset.name} to the current Graphics Settings.\nRight now {pipelineAsset.name} is assigned. Do you want to replace it?";
+                    buttonText = "Replace";
+                }
+
+                if (GraphicsSettings.defaultRenderPipeline != (RenderPipelineAsset)target)
```

Comment:
You already had the cast in `asset`, you can also extract the cast outside of both buttons.